lib/config/presets/index.ts

```diff
@@ -222,15 +227,33 @@ export async function getPreset(
   const { presetSource, repo, presetPath, presetName, tag, params } =
     parsePreset(preset);
   const cacheKey = `preset:${preset}`;
-  let presetConfig = memCache.get<Preset | null | undefined>(cacheKey);
+  const presetCache = GlobalConfig.get('presetCache', false);
+
+  let presetConfig: Preset | null | undefined;
+
+  if (presetCache) {
+    presetConfig = await packageCache.get(presetCacheNamespace, cacheKey);
+  } else {
+    presetConfig = memCache.get(cacheKey);
+  }
+
   if (is.nullOrUndefined(presetConfig)) {
     presetConfig = await presetSources[presetSource].getPreset({
       repo,
       presetPath,
       presetName,
       tag,
     });
-    memCache.set(cacheKey, presetConfig);
+    if (presetCache) {
+      await packageCache.set(
+        presetCacheNamespace,
+        cacheKey,
+        presetConfig,
+        getTtlOverride(presetCacheNamespace) ?? 15,
```

Comment:
the current implementation causes a ttl reset whenever the preset is used again. so if there is never a a timespan big enough we will always use the cached value without a hard timeout.

Reply:
I'm trying to understand what do you mean by ttl reset @viceice. the workflow in my understanding is like the following, would you mind tell me how it happens? 

1. no cache,  load the cache from source
2. set the cache with ttl
3. renovate's been using the cached version of preset within the ttl
4. ttl expires
5. back to step 1 load the source again.

Reply:
on step 3 the ttl is reset because we save the cached value again to the cache, so it doesn't expire if we call this function faster than the ttl can expire

Reply:
> because we save the cached value again to the cache

Didn't the implementation calls the `packageCache.set` only when there is no `presetConfig` return from `packageCache.get` previously, which only happens at the begining or on ttl expires? 

Unless the `packageCache.get` does somehow refreshes the ttl on invocation? I couldn't find it in neither `packageCache.get` nor the `file` and `redis` cache proxies.

Reply:
you're right, i didn't see the indention on mobile 🙃